frontend/controller/service-worker.js

```diff
@@ -98,6 +98,10 @@ sbp('sbp/selectors/register', {
       ])
       const swRegistration = await navigator.serviceWorker.register(`/assets/js/sw-primary.js?${params}`, { scope: '/' })
 
+      if (swRegistration.active == null && swRegistration.installing == null && swRegistration.waiting == null) {
+        throw new Error('No valid service worker found')
```

Comment:
Nit: o4-mini (high) recommends adding the scope: `No valid service worker found (scope: ${swRegistration.scope})`

Reply:
Not sure how that's relevant here, since there's only one scope